marketplace/neo4j.template.yaml

```diff
@@ -424,15 +424,33 @@ Resources:
             - "install_neo4j_from_yum() {\n"
             - "  echo \"Installing Graph Database...\"\n"
             - "  export NEO4J_ACCEPT_LICENSE_AGREEMENT=yes\n"
-            - "  if ! curl --fail http://versions.neo4j-templates.com/target.json; then\n"
-            - "    echo 'Failed to resolve Neo4j version from http://versions.neo4j-templates.com, using latest'\n"
-            - "    local -r graphDatabaseVersion='neo4j-enterprise'\n"
+            - "  ASG_NAME=$(aws --region $region cloudformation describe-stack-resources --stack-name $stackName --query 'StackResources[?ResourceType==`AWS::AutoScaling::AutoScalingGroup`].PhysicalResourceId' --output text)\n"
+            - "  NEO4J_VERSION_ASG_TAG=$(aws --region $region autoscaling describe-auto-scaling-groups --auto-scaling-group-names $ASG_NAME --query 'AutoScalingGroups[0].Tags[?Key==`NEO4J_VERSION`].Value' --output text)\n"
+            - "  if [[ ! -z $NEO4J_VERSION_ASG_TAG ]]; then\n"
+            - "    echo \"Found NEO4J_VERSION_ASG_TAG Tag on AutoScalingGroup: NEO4J_VERSION_ASG_TAG=$NEO4J_VERSION_ASG_TAG\"\n"
+            - "    PACKAGE_VERSION=$NEO4J_VERSION_ASG_TAG\n"
+            - "    local -r graphDatabaseVersion=\"neo4j-enterprise-$PACKAGE_VERSION\"\n"
             - "  else\n"
-            - "    local -r graphDatabaseVersion=\"neo4j-enterprise-$(curl http://versions.neo4j-templates.com/target.json | jq -r '.aws.\"5\"')\"\n"
+            - "    PACKAGE_VERSION=$(curl --fail http://versions.neo4j-templates.com/target.json | jq -r '.aws.\"5\"' || echo \"\")\n"
+            - "    if [[ ! -z $PACKAGE_VERSION ]]; then\n"
+            - "      echo \"Found PACKAGE_VERSION from http://versions.neo4j-templates.com : PACKAGE_VERSION=$PACKAGE_VERSION\"\n"
+            - "      local -r graphDatabaseVersion=\"neo4j-enterprise-$PACKAGE_VERSION\"\n"
+            - "    else\n"
+            - "      echo 'Failed to resolve Neo4j version from http://versions.neo4j-templates.com, using PACKAGE_VERSION=latest'\n"
+            - "      PACKAGE_VERSION=\"latest\"\n"
+            - "      local -r graphDatabaseVersion='neo4j-enterprise'\n"
+            - "    fi\n"
             - "  fi\n"
             - "  yum -y install \"${graphDatabaseVersion}\"\n"
             - "  yum update -y aws-cfn-bootstrap\n"
             - "  systemctl enable neo4j\n"
+            - "  if [[ \"$PACKAGE_VERSION\" == \"latest\" ]]; then\n"
```

Comment:
we do not need the if's on line 447 and 450 ...we can execute the respective if operations in the else starting on line 433 since both the if conditions are satisfied in the else

Reply:
So the `if` condition on line 447 cannot be placed in the `else` condition on line 433, as the command `/user/share/neo4j/bin/neo4j --version` is needed to determined the installed version of neo4j. This installation happens on line 444

Also the `if` condition on line 450 is checking to see if the ASG needs to be tagged. This could be moved to the `else` condition on line 433, however, I would prefer the installation of neo4j to take place first, i.e. line 444, and then once neo4j is installed, then tag the ASG with the installed version

Reply:
i agree with the reasoning for the first if ...for the second condition even though neo4j installation is done it does not necessarily mean neo4j is up since its not started...so if u tag it after the yum install it is similar to tagging it before yum install ...it will help you reduce one if condition...let me know what you think about it

Reply:
or maybe u wanna do the tagging after neo4j starts completely

Reply:
On second thought, do you think it may be ok to leave the tagging logic where it is, if something fails in the script below, it may still be nice to be able to see the version tag on the ASG whilst debugging the deployment. Given that neo4j has been installed, whether it has started or not does not affect the accuracy of the information provided by the tag

Also, from a readable point of view, the variable `NEO4J_VERSION_ASG_TAG` is used within the same method. It is possible to reference this variable towards the end of the script in the `start_neo4j` method, but this requires the reader to have to keep track of this variable across the file.

What do you think?